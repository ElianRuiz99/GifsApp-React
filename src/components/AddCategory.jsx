import { useState } from "react";

export const AddCategory = ({ onNewCategory }) =>{

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        // console.log( event.target.value )
        setInputValue( event.target.value );
    }

    const onSubmit= (event) => {
        event.preventDefault();
        // console.log(inputValue)
        if( inputValue.trim().length <= 1 ) return;

        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={ e => onSubmit(e) }>
           <input 
                type="text" 
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ (e) => onInputChange(e) }
            /> 
        </form>
    );
}