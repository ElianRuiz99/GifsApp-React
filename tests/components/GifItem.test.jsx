import { render, screen } from "@testing-library/react";
import { scryRenderedComponentsWithType } from "react-dom/test-utils";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en el componente de GifItem.jsx', () => {  

    const title = "Adventure Time"
    const url = "https://adventure/unaImagen.jpg"

    test('debe de hacer match con el snapshot', () => { 

        const {container} = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el Alt correspondiente', () => { 

        render( <GifItem title={ title } url={ url } /> );
        screen.debug();

        expect( screen.getByRole('img').src ).toBe( url );
        expect( screen.getByRole('img').alt ).toBe( title );
        
    });

    test('Debe de mostrar el titulo', () => { 

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();  

    });

})