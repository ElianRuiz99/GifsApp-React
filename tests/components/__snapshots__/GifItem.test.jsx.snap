// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de GifItem.jsx debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Adventure Time"
      src="https://adventure/unaImagen.jpg"
    />
    <p>
      Adventure Time
    </p>
  </div>
</div>
`;
